### PR TITLE
feat: add simple search param on search

### PR DESCRIPTION
**Pull request by istvanmakary**
This Pull Request was created to simplify how search is handled in the Select component. 
To cover most of the use cases, there is no need to store the current search value and do the filtering, the Select module does it for you. If you need advance filtering you can specify your search function with any custom logic.
For really complex cases the previous implementation should be used.
Of course it's backward compatible.

## ✅ Checklist

- [x] All changes are covered by tests
- [x] The changes are covered in docs
- [x] The commits follow conventional commits standards

### CODE REVIEW

File: packages/docs/liveEditorCode/select.code.js [hunk 1]
```diff
@@ -18,10 +16,9 @@
       disabled={false}
       onChange={v => setSelected(v)}
       required={false}
-      searchValue={searchValue}
       searchPlaceholder="searchplaceholder"
-      onSearchChange={v => setSearchValue(v)}
       dropdownUp={false}
+      search={true}
```
Comment (sbrandwoo) [inline packages/docs/liveEditorCode/select.code.js]: In the documentation, is it possible to document what parameters the function takes?
https://transferwise.github.io/neptune-web/branch/improve-select-search/components/Select

File: packages/components/src/select/Select.js [hunk 2]
```diff
@@ -44,6 +47,9 @@ const BOOTSTRAP_DROPDOWN_ANIMATION_TIME = 200;
 
 const defer = fn => setTimeout(fn, 0);
 
+const defaultFilterFunction = (option, keyword) =>
+  option.label && option.label.toLowerCase().includes(keyword.toLowerCase());
```
[the remark this answers is not in the document]
Comment (istvanmakary) [reply packages/components/src/select/Select.js]: done

File: packages/docs/liveEditorCode/select.code.js (hunk 1 above)
Comment (istvanmakary) [reply packages/docs/liveEditorCode/select.code.js]: done

File: packages/components/src/select/Select.js [hunk 3]
```diff
@@ -44,6 +47,12 @@ const BOOTSTRAP_DROPDOWN_ANIMATION_TIME = 200;
 
 const defer = fn => setTimeout(fn, 0);
 
+const includesString = (str1, str2) => str1.toLowerCase().includes(str2.toLowerCase());
```
Comment (sbrandwoo) [inline packages/components/src/select/Select.js]: Does `includes` get polyfilled for IE?

File: packages/components/src/select/Select.js [hunk 4]
```diff
@@ -44,6 +47,12 @@ const BOOTSTRAP_DROPDOWN_ANIMATION_TIME = 200;
...
+
+const defaultFilterFunction = (option, keyword) =>
```
Comment (sbrandwoo) [inline packages/components/src/select/Select.js]: This filters out headers too. I think that's a good thing though.

File: packages/components/src/select/Select.js (hunk 3 above)
Comment (andreapiras) [reply packages/components/src/select/Select.js]: Can we not use indexOf here?

Comment (istvanmakary) [reply packages/components/src/select/Select.js]: sure, I'll change it!

File: packages/components/src/select/Select.js (hunk 4 above)
Comment (istvanmakary) [reply packages/components/src/select/Select.js]: As I saw, the default behaviour was to filter out headers. Or not?

File: packages/components/src/select/Select.js (hunk 3 above)
Comment (istvanmakary) [reply packages/components/src/select/Select.js]: Done!

File: packages/components/src/select/Select.js (hunk 4 above)
Comment (istvanmakary) [reply packages/components/src/select/Select.js]: We discussed with @sbrandwoo that the current implementation of this feature is fine.

**Pull request merged**